.github/workflows/build-beta-app.yml

```
@@ -0,0 +1,68 @@
+name: "Beta build"
+
+on:
+  push:
+    branches:
+      - main
+  merge_group:
+  pull_request:
+    branches:
+      - main
+    types: [ opened, synchronize ]
+  workflow_call:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  code-analysis:
+    uses: ./.github/workflows/code-analysis.yml
+  unit-tests:
+    uses: ./.github/workflows/gradle-run-unit-tests.yml
+  build-app:
+    needs: [ unit-tests, code-analysis ]
```

Maybe we should also run the UI tests from here?

```suggestion
  ui-tests:
    uses: ./.github/workflows/gradle-run-ui-tests.yml
  build-app:
    needs: [ ui-tests, unit-tests, code-analysis ]
```

OR, merge UI-tests together with unit tests.
And we could name it `test`, which has two jobs: `unit-tests` and `ui-tests`.

Similar to `code-analysis` (lint and style).